daemon/CMakeLists.txt

```diff
@@ -1,10 +1,12 @@
-# Create a target that builds the ebpf program
-add_ebpf_program_target(bpftime_daemon_ebpf_target ${CMAKE_CURRENT_SOURCE_DIR}/kernel/bpf_tracer.bpf.c ${CMAKE_CURRENT_BINARY_DIR}/bpf_tracer.bpf.o)
+if(${BPFTIME_BUILD_WITH_LIBBPF})
```

Comment:
bpftime_daemon won't work on platforms other than Linux. The whole daemon folder could be disabled. It relies on kernel eBPF to work. 
- bpftime_daemon acts similarly like syscall server, but it works by monitoring syscalls like `bpf` in kernel by kernel eBPF and sends such calling records to user space and stores it to shared memory. While syscall server hook such calls in userspace

Reply:
noted, will do that change.